crates/common/src/utils.rs

```diff
@@ -38,3 +44,56 @@ pub fn erc7201(id: &str) -> B256 {
     let x = U256::from_be_bytes(keccak256(id).0) - U256::from(1);
     keccak256(x.to_be_bytes::<32>()) & B256::from(!U256::from(0xff))
 }
+
+/// Returns the contract name given the artifact path.
+pub fn find_target_name(output: &ProjectCompileOutput, target_path: &Path) -> Result<String> {
+    let names = output
+        .artifact_ids()
+        .filter(|(id, _)| id.source == target_path)
+        .map(|(id, _)| id.name)
+        .collect::<Vec<_>>();
+
+    if names.len() > 1 {
```

Comment:
if we use additional compiler profiles / restrictions there'll be multiple names for the same contract found and will bail with `Multiple contracts found in the same file` even if there's a single contract. For example for
```toml
optimizer = true

additional_compiler_profiles = [
  { name = "another", optimizer_runs = 10 },
]
compilation_restrictions = [
  { paths = "src/another/AnotherCounter.sol", optimizer_runs = 10 },
]
```

where `contract AnotherCounter is Counter`  there'll be `Counter.another.json` and `Counter.json` generated hence `forge inspect src/counter/Counter.sol abi` will fail with `Error: Multiple contracts found`